src/backend/joanie/core/api/client.py

```diff
@@ -603,19 +604,37 @@ class CertificateViewSet(
         "enrollment__course_run__course",
     )
 
-    def get_queryset(self):
-        """
-        Custom queryset to get user certificates
-        """
-        queryset = super().get_queryset()
-        username = (
+    def get_username(self):
+        """Get the authenticated username from the request."""
+        return (
             self.request.auth["username"]
             if self.request.auth
             else self.request.user.username
         )
-        return queryset.filter(
-            Q(order__owner__username=username) | Q(enrollment__user__username=username),
-        ).distinct()
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.action == "list":
+            queryset = queryset.distinct()
+
+        return queryset
```

Comment:
Could we assume to use `distinct` on the class queryset in order to remove this method override?

Reply:
I don't see the point to add a distinct on the `retrieve` action.